packages/web-components/src/drawer/drawer.ts

```diff
@@ -0,0 +1,461 @@
+import { attr, css, FASTElement, observable, Updates } from '@microsoft/fast-element';
+import { isTabbable } from 'tabbable';
+import { eventAnimationEnd, keyEscape, keyTab } from '@microsoft/fast-web-utilities';
+import { colorNeutralStroke1, colorTransparentStroke } from '../theme/design-tokens.js';
+import { DrawerModalType, DrawerPosition, DrawerSize, DrawerType } from './drawer.options.js';
+
+export class Drawer extends FASTElement {
+  private closing: boolean = false;
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    document.addEventListener('keydown', this.handleDocumentKeydown);
+
+    Updates.enqueue(() => {
+      this.updateTrapFocus();
+      this.setComponent();
+    });
+  }
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+    document.removeEventListener('keydown', this.handleDocumentKeydown);
+    this.updateTrapFocus(false);
+  }
+
+  /**
+   * The content container.
+   * @public
+   * @remarks
+   * HTML Attribute: content
+   */
+  @observable
+  public content?: HTMLElement;
+
+  /**
+   * The dialog element.
+   * @public
+   *
+   */
+  @observable
+  public dialog!: HTMLDialogElement;
+
+  /**
+   * Determines whether the drawer should be displayed as modal, non-modal, or alert.
+   * When in modal or alert mode, an overlay is applied over the rest of the view.
+   * @public
+   * @remarks
+   * HTML Attribute: modalType
+   */
+  @attr({ attribute: 'modal-type' })
+  public modalType: DrawerModalType = DrawerModalType.modal;
+
+  /**
+   * The ID of the element that labels the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-labelledby
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  public ariaLabelledby?: string;
+
+  /**
+   * The ID of the element that describes the drawer.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-describedby
+   */
+  @attr({ attribute: 'aria-describedby' })
+  public ariaDescribedby?: string;
+
+  /**
+   * Sets the type of the drawer (overlay/inline).
+   * @public
+   * @remarks
+   * HTML Attribute: type
+   * @defaultValue overlay
+   */
+  @attr
+  public type?: DrawerType = DrawerType.overlay;
+
+  /**
+   * Sets the position of the drawer (start/end).
+   * @public
+   * @remarks
+   * HTML Attribute: position
+   * @defaultValue end
+   */
+  @attr
+  public position?: DrawerPosition;
```

Comment:
You show in the comment that the default value is `end`. How does this play out in your code? I see that position has a type but no default value set here. Same with `size`.

Reply:
Those attributes are used for applying CSS styles. The default style of the control is set up to handle cases where these attributes are not explicitly defined. This means that if the size attribute, for instance, isn't set, the control will automatically adopt the default styling.

Example..

```html
// medium ( default ) size

<card></card>
:host {
  width: 200px; 
}

// Small size

<card size="small"></card>
:host([size='small']) {
  width: 100px;
}

// Large size

<card size="large"></card>
:host([size='large']) {
  width: 400px;
}
```

Reply:
Thx for clarifying